### PR TITLE
Added an example that uses datasets from MOABB and the RG+QuantumSVM pipeline

### PR DESCRIPTION
**Pull request by toncho11**
The example visualizes and compares with two other "standard" pipelines. 
It depends on the newest version of MOABB 0.4.6.

### CODE REVIEW

File: examples/ERP/classify_P300_bi.py [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    def __init__(self):
+        pass
+
+    def fit(self, X, y):
+        """fit."""
+        return self
+
+    def transform(self, X):
+        """transform. """
+        return np.reshape(X, (X.shape[0], -1))
```
Comment (gcattan) [inline examples/ERP/classify_P300_bi.py:66]: If we intend to reuse this, can you add it to `utils.filtering`?

File: examples/ERP/classify_P300_bi.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+overwrite = True  # set to True if we want to overwrite cached results
+
+pipelines = {}
+
+# new pipeline provided by pyRiemann-qiskit
+pipelines["RG+Quantum"] = QuantumClassifierWithDefaultRiemannianPipeline(
+    shots=None,  #'None' forces classic SVM
+    nfilter=2,  #default 2
+    dim_red=PCA(n_components=10), #default 10, higher values render better performance with the SVM version used in qiskit
+    #q_account_token="" #IBM Quantum TOKEN
```
Comment (gcattan) [inline examples/ERP/classify_P300_bi.py:100]: Can you raise a defect for the `q_account_token`, following your observations in our Skype conversation?

File: examples/ERP/classify_P300_bi.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+    TangentSpace(),
+    PCA(n_components=10),
+    LDA(solver="lsqr", shrinkage="auto"),
+)
+
+# standard pipeline 2
+pipelines["Xdw+LDA"] = make_pipeline(
+    Xdawn(nfilter=2, estimator="scm"), 
+    Vectorizer(), 
+    LDA(solver="lsqr", shrinkage="auto")
```
Comment (gcattan) [inline examples/ERP/classify_P300_bi.py:119]: You do not have the same number of component in your feature vector, as compared with the two other pipelines. Is this intended?

File: examples/ERP/classify_P300_bi.py [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    palette="Set1",
+)
+sns.pointplot(data=results, y="score", x="pipeline", 
+              ax=ax, zorder=1, 
+              palette="Set1")
+
+ax.set_ylabel("ROC AUC")
+ax.set_ylim(0.3, 1)
+
+fig.show()
```
Comment (gcattan) [inline examples/ERP/classify_P300_bi.py:158]: Can you post on a comment here the resulting figure?

File: examples/ERP/classify_P300_bi.py [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+# bi2012,bi2013a, bi2014a, bi2014b, bi2015a, bi2015b, BNCI2014009
+import moabb 
+from moabb.datasets import bi2012 
+
+from moabb.evaluations import WithinSessionEvaluation
+from moabb.paradigms import P300
+from pyriemann_qiskit.classification import QuantumClassifierWithDefaultRiemannianPipeline
+from sklearn.decomposition import PCA
+
+
```
Comment (gcattan) [inline examples/ERP/classify_P300_bi.py:41]: Add here `print(_doc_)`

**Review by gcattan — COMMENTED** [review]
Good job :) Can you in addition, add this example to `doc/index.rst`?

image

File: examples/ERP/classify_P300_bi.py [hunk 6]
```diff
@@ -0,0 +1,169 @@
+"""
+====================================================================
+ERP EEG decoding with Quantum Classifier.
+====================================================================
+
+It uses QuantumClassifierWithDefaultRiemannianPipeline. This pipeline uses
+Riemannian Geometry and Tangent Space to generate features and a quantum SVM
+classifier. It uses MOABB for the evaluation and compariosn with 2 other
+standard pipelines.
```
Comment (gcattan) [inline examples/ERP/classify_P300_bi.py:9]: Adapt here text according to our Skype conversation.

File: examples/ERP/classify_P300_bi.py [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    # default n_components=10, a higher value renders better performance with
+    # the SVM version used in qiskit
+    dim_red=PCA(n_components=10), 
+    # q_account_token='' #IBM Quantum TOKEN
+    )
+
+# Here we provide some pipelines for comparison:
+
+# Standard SVM on raw data
+from sklearn import svm
```
Comment (gcattan) [inline examples/ERP/classify_P300_bi.py:109]: Move this at the beginning.

File: examples/ERP/classify_P300_bi.py (hunk 1 above)
Comment (gcattan) [reply examples/ERP/classify_P300_bi.py]: Do we still need this since you changed your example?

File: examples/ERP/classify_P300_bi.py (hunk 2 above)
Comment (gcattan) [reply examples/ERP/classify_P300_bi.py]: Follow-up opened here: https://github.com/pyRiemann/pyRiemann-qiskit/issues/40

File: doc/requirements.txt [hunk 8]
```diff
@@ -10,4 +10,5 @@ pandas
 cvxpy==1.1.12
 qiskit==0.20.0
 scipy==1.7.3
+moabb>=0.4.6
```
[the remark this answers is not in the document]
Comment (gcattan) [reply doc/requirements.txt:13]: Looks like moabb is not compatible with Python 3.6.
We can try to install from the source as you suggested or drop support for python 3.6 and eventually add support for 3.9 (but qiskit-aer is not compatible with 3.9, so we can only support 3.9 after updating qiskit).

[the remark this answers is not in the document]
Comment (gcattan) [reply doc/requirements.txt]: Ok, so I tried installing from the source, but it fails with the same error.

**Review by gcattan — APPROVED** [review]

File: doc/requirements.txt (hunk 8 above)
[the remark this answers is not in the document]
Comment (sylvchev) [reply doc/requirements.txt:13]: Yes, we dropped support of 3.6 when it was deprecated. You could move to python-3.7 for base requirement (support will end 27 Jun 2023).
Good idea to move moabb dependency to doc, no need to have it in the basic requirements.

File: examples/ERP/classify_P300_bi.py [hunk 9]
```diff
@@ -0,0 +1,144 @@
+"""
+====================================================================
+ERP EEG decoding with Quantum Classifier.
```
Comment (sylvchev) [inline examples/ERP/classify_P300_bi.py:3]: You could change the title, as this is the same title than `plot_classify_EEG_quantum.py` and they should appears dfferently on the doc.

File: examples/ERP/classify_P300_bi.py [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+sns.pointplot(data=results,
+              y="score",
+              x="pipeline",
+              ax=ax, zorder=1,
+              palette="Set1")
+
+ax.set_ylabel("ROC AUC")
+ax.set_ylim(0.3, 1)
+
+fig.show()
```
Comment (sylvchev) [inline examples/ERP/classify_P300_bi.py:144]:
```suggestion
plt.show()
```

The figure is not shown on the generated documentation, try using `plt.show()` to display the plot.

**Review by sylvchev — CHANGES_REQUESTED** [review]

File: doc/requirements.txt (hunk 8 above)
[the remark this answers is not in the document]
Comment (gcattan) [reply doc/requirements.txt:13]: Ok, I removed support for python-3.6, and opened issue #43 as a reminder to bring support for 3.9.

**Comment by toncho11** [comment]
Hi,

I think everything is ready for this PR to be accepted.

**Review by qbarthelemy — APPROVED** [review]

**Comment by gcattan** [comment]
@sylvchev did you have a chance to review the last modification?

**Comment by gcattan** [comment]
@sylvchev

**Review by sylvchev — APPROVED** [review]

**Comment by gcattan** [comment]
@toncho11 thanks for taking care of this. Do you think we can reuse some elements of your presentation at `nbt-berlin` to improve our documentation?

**Comment by gcattan** [comment]
I was wondering how can provides an estimate of the chance level for the bi2012 dataset. Is there something we can reuse from moabb @sylvchev ?

**Pull request merged**